Optimize `entity.add()`.

player-03 opened the pull request:
Instead of attempting to add an entity to ALL views, it's best to check only views of the newly-added component. So if you add a `String` component, the function will call `ViewsOfComponentString.inst().addIfMatched()` instead of iterating through `Workflow.views`.

Interestingly, `entity.remove()` was already optimized this way, so I assume this was an oversight.

player-03 commented:
Also, any chance you could turn on Issues for this repo? I have questions/comments, but there's nowhere to put them.

It'll be on the [settings page](https://github.com/onehundredfeet/hcqe/settings), under "features."

onehundredfeet commented:
Sorry, I totally missed this. I'll turn on issues.

onehundredfeet commented:
I added the features. I will look at your PR. 

I also just renamed the depot.  Sorry if that causes you any issues.

onehundredfeet commented:
Would you mind resolving against the changes? There's been so many it's hard to see through it.

player-03 committed:
Optimize `entity.add()`.

player-03 commented:
Finally done, I think.

player-03 commented:
Ok, but for the record, [the issue is still very much present](https://github.com/onehundredfeet/hmecs/blob/0946fb8b2a3544fe26bbf3f97f1319bd1e6b04d3/src/ecs/Entity.hx#L166).

```haxe
macro if (__entity__.isActive()) {
	for (v in ecs.Workflow.views) { // <- inefficient
		@:privateAccess v.addIfMatched(__entity__);
	}
}
```

You're iterating through _all_ views each time you add _any_ component, despite the fact that most of those views don't even include that component and therefore couldn't possibly be affected.

onehundredfeet commented:
Agreed. Nice to hear from you again.

I haven't had a lot of time to look at this but I've come into some more time now and was going to take a look.

I'm not sure why it said I closed the issue?  I didn't touch the issue status.

player-03 commented:
Oh, ok. I'll put it back then.

Whatever happened, it prompted me to look at your latest change, which I'm actually considering using. So that's good.

player-03 committed:
Clarify `Entity.add()`, and fix a bug.

You don't use `return` during array comprehension; that will make the enclosing function return. Other than that, I just think it's easier to read without chains of `concat()` and `map()`.

onehundredfeet commented:
huh? why is it auto closing?

onehundredfeet commented:
Reopening.

onehundredfeet commented:
All the changes disappeared, what is going on?

player-03 commented:
I'll open a new one.

onehundredfeet commented:
I was browsing your branch.  You've done an impressive amount of updates.

player-03 commented:
Thanks. Hope you can find some useful stuff in there. And I'm happy to explain if you find anything confusing.

player-03 commented:
Betting it broke because player-03/echoes is no longer a fork. So I sent the new pull request from my forked repo instead.
